wagtail/wagtailimages/models.py

```diff
@@ -71,6 +61,20 @@ class AbstractImage(models.Model, TagSearchable):
     focal_point_width = models.PositiveIntegerField(null=True, blank=True)
     focal_point_height = models.PositiveIntegerField(null=True, blank=True)
 
+    @staticmethod
+    def get_upload_to(instance, filename):
```

Comment:
Don't see why not, I'll get something over to you in the next couple of days? (Bit busy at the moment!)